### PR TITLE
Raytune hpo example walkthrough

### PR DESCRIPTION
**Pull request by ncau**
Example walkthrough of how to run the HPO demos

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+![](./docs/01.png)
+
+* Log in, then go to Data Science Projects and create a project:  
+![](./docs/02.png)
+
+* Once the project is created, click on Create a workbench:<br/><br/>
+![](./docs/03.png)
+
+* Add a cluster storage:
+![](./docs/04.png)
```
Comment (kusumachalasani) [inline examples/README.md]: The image shows 'Data Connections' tab when it talks about cluster storage

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+* Sufficient worker nodes for your configuration(s) with Nvidia GPUs (Ampere-based recommended);
+* An AWS S3 bucket to store experimentation results.
+
+## Setup
+
+* Access the OpenShift AI dashboard, for example from the top navigation bar menu:
+![](./docs/01.png)
+
+* Log in, then go to Data Science Projects and create a project:  
+![](./docs/02.png)
```
Comment (kusumachalasani) [inline examples/README.md:18]: docs has 02png.png

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# HPO with raytune POC on OpenShift AI
+
+This example demonstrates how to use raytune for hyperparameter optimisation using raytune on OpenShift AI.
```
Comment (dinogun) [inline examples/README.md]: Would be good to explain the HPO example a bit more. What is the objective of this HPO exercise, what is the base example, what parameters are being tuned and what is the expected outcome that we are aiming for.

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    * Raytune-oai-demo-mlmd.ipynb
+    * Raytune-oai-MR-gRPC-demo.ipynb
+
+* In the openshift console open “copy login token”:  
+![](./docs/12.png)
+
+* Replace the TOKEN and SERVER variables with those from the login command:
+![](./docs/13.png)
+
+* Restart and run the kernel, the results should be generated.
```
Comment (dinogun) [inline examples/README.md]: Would be good to add a brief summary of what was achieved here

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+* Clone the following repository (https://github.com/kruize/hpo-poc.git):
+![](./docs/09.png)
+![](./docs/10.png)
+
+* Navigate to the demos folder (hpo-poc/demos/)
+
+* Repeat the following steps for each of the three files:
+    * Raytune-oai-demo-mlmd.ipynb
+    * Raytune-oai-demo-mlmd.ipynb
+    * Raytune-oai-MR-gRPC-demo.ipynb
```
Comment (dinogun) [inline examples/README.md]: Would be good to add a brief on what additional functionality is available in the other notebooks

**Review by kusumachalasani — COMMENTED** [review]
Can you also include, that we have hardcoded the datatype ids created in https://github.com/ncau/hpo-poc/blob/hpoExample/demos/raytune-oai-MR-gRPC-demo.ipynb related to HPO

```
# Use the default metadata types for Model and other metadata types that are part of the metadata database.
# Replace these values based on the MR database using the code from previous block.
model_type_id = 12
hpo_config_type_id = 22
hpo_experiment_type_id = 23
hpo_trial_type_id = 24
data_type_id = 25
trainer_type_id = 26
metrics_type_id = 27
```

File: examples/README.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+* Then create a workbench with the following settings:  
+![](./docs/06.png)
+
+* Add data connection with relevant details of your S3 storage bucket:
+![](./docs/07.png)
+
+### Setting Up Model Registry
+
+To install the Model Registry Controller, follow these steps:
```
Comment (dinogun) [inline examples/README.md:36]: Why is the model registry needed, would be good to explain that a bit here

**Review by kusumachalasani — APPROVED** [review]
LGTM!

File: examples/README.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+* Clone the following repository (https://github.com/kruize/hpo-poc.git):
+![](./docs/09.png)
+![](./docs/10.png)
+
+* Navigate to the demos folder (hpo-poc/demos/)
+
+* Repeat the following steps for each of the three files:
+    * Raytune-oai-demo.ipynb - base demo
+    * Raytune-oai-demo-mlmd.ipynb - stores and retrives all data in a mlmd store
+    * Raytune-oai-MR-gRPC-demo.ipynb - stores and retreives all data using the model registry api
```
Comment (dinogun) [inline examples/README.md]: retrieves

File: examples/README.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
+# HPO with raytune POC on OpenShift AI
+
+This example primarily focuses on optimizing machine learning models using RayTune. It demonstrates the process of hyperparameter tuning to find the best-performing model configuration for a given model, leveraging the capabilities of RayCluster and CodeFlare. There are three examples provided, all of these have hidden layer size and learning rate as hyperparameters. The base demo will simply upload the model to the s3 bucket. The mlmd demo creates and stores the model into a mlmd store and the MR-gRPC using the MR API to do the same. 
+
+> [!IMPORTANT]
+> This example has been tested with the configurations listed in the [validation](#validation) section.
```
Comment (dinogun) [inline examples/README.md]: Is there a validation section here?

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**